edx_toggles/toggles/state.py

```diff
@@ -0,0 +1,265 @@
+"""
+Toggle state report API.
+"""
+from collections import OrderedDict
+
+from django.conf import settings
+from edx_django_utils.monitoring import get_code_owner_from_module
+from waffle.models import Flag, Switch
+
+from . import SettingDictToggle, SettingToggle, WaffleFlag, WaffleSwitch
+
+
+class ToggleStateReport:
+    """
+    Convenient class to collect toggle objects across a codebase and expose them via a REST API.
+
+    Many methods are overridable to expose custom toggle objects in different IDAs.
+
+    Use as follows:
+
+        report = ToggleStateReport().as_dict()
+    """
+
+    def as_dict(self):
+        """
+        Produce a JSON-convertible report dict.
+
+        Return:
+            report (OrderedDict): this contains following keys: "waffle_flags", "waffle_switches", "django_settings".
+        """
+        report = OrderedDict()
+        report["waffle_flags"] = sorted_values_by_name(self.get_waffle_flags())
+        report["waffle_switches"] = sorted_values_by_name(self.get_waffle_switches())
+        report["django_settings"] = sorted_values_by_name(self.get_django_settings())
+        return report
+
+    def get_waffle_flags(self):
+        """
+        Get all waffle flags and their state, indexed by name.
+        """
+        flags_dict = {}
+        self.add_waffle_flag_instances(flags_dict)
+        self.add_waffle_flag_state(flags_dict)
+        self.add_waffle_flag_computed_status(flags_dict)
+        return flags_dict
+
+    def add_waffle_flag_instances(self, flags_dict):
+        """
+        Add waffle flag instances, indexed by name.
+        """
+        _add_waffle_flag_instances(flags_dict)
```

Comment:
Curious of the benefit of all the single line functions, rather than just inlining the function details?

Reply:
It makes it clear that the implementation does not depend on `self`. Granted, this is a not a huge benefit. But I like the fact that the base ToggleStateReport class is kept relatively small in terms of lines of code.

Reply:
Thanks. Just curious.